core/src/main/java/io/openepcis/epc/eventhash/ContextNode.java

```diff
@@ -193,24 +193,24 @@ && getValue() != null
       }
 
       return preHashBuilder.toString();
-    } else if (children.isEmpty() && getName() != null && getValue() != null && !TemplateNodeMap.isEpcisField(this) && EventHashGenerator.getCbvVersion().equals(CBVVersion.VERSION_2_1_0)) {
+    } else if (children.isEmpty() && getName() != null && getValue() != null && !TemplateNodeMap.isEpcisField(this) && cbvVersion.equals(CBVVersion.VERSION_2_1_0)) {
```

Comment:
It's okay to allow null cbvVersion - but please switch condition to CBVVersion.VERSION_2_1_0.equals(cbvVersion)